AI/VCAI/VCAI.cpp

```diff
@@ -590,6 +599,37 @@ void VCAI::showBlockingDialog(const std::string &text, const std::vector<Compone
 	});
 }
 
+void VCAI::showTeleportDialog(TeleportChannelID channel, std::vector<ObjectInstanceID> exits, bool impassable, QueryID askID)
+{
+	LOG_TRACE_PARAMS(logAi, "askID '%i', exits '%s'", askID % exits);
+	NET_EVENT_HANDLER;
+	status.addQuery(askID, boost::str(boost::format("Teleport dialog query with %d exits")
+																			% exits.size()));
+
+	ObjectInstanceID choosenExit;
+	if(impassable)
+		knownTeleportChannels[channel]->passability = TeleportChannel::IMPASSABLE;
+	else
+	{
+		if(destinationTeleport != ObjectInstanceID() && vstd::contains(exits, destinationTeleport))
+			choosenExit = destinationTeleport;
+
+		if(!status.channelProbing())
+		{
+			vstd::erase_if(exits, [&](ObjectInstanceID id) -> bool
+			{
+				return vstd::contains(visitableObjs, cb->getObj(id)) || id == choosenExit;
+			});
+			teleportChannelProbingList = exits;

```

Comment:
You should move exits parameter to teleportChannelProbingList or accept exits by const reference